Documentation/src/case-studies/calendar-code.md

```diff
@@ -0,0 +1,110 @@
+﻿
+## Program.cs
+
+The complete calendar program:
+
+```cs
+using System.Collections.Immutable;
+using Funcky.Extensions;
+using System.Globalization;
+using Funcky;
+using Funcky.Monads;
+using static Funcky.Functional;
+
+namespace Calendar;
+
+internal static class StringExtensions
+{
+    public static string Center(this string text, int width)
+        => (width - text.Length) switch
+        {
+            0 => text,
+            1 => $" {text}",
+            _ => Center($" {text} ", width),
+        };
+}
+
+internal class Program
+{
+    private const int FirstDayOfTheWeek = 0;
+    private const int HorizontalMonths = 3;
+    private const int DaysInAWeek = 7;
+    private const int WidthOfDay = 3;
+    private const int WidthOfAWeek = WidthOfDay * DaysInAWeek;
+    private const string MonthNameFormat = "MMMM";
+
+    private static void Main(string[] args)
+        => Console.WriteLine(CreateCalendarString(ExtractYear(args).GetOrElse(DateTime.Now.Year)));
+
+    private static Option<int> ExtractYear(IEnumerable<string> args)
+        => args.FirstOrNone()
+            .AndThen(ParseExtensions.ParseInt32OrNone);
+
+    private static string CreateCalendarString(int year)
+        => Sequence.Successors(JanuaryFirst(year), NextDay)
+            .TakeWhile(SameYear(year))
+            .AdjacentGroupBy(day => day.Month)
+            .Select(LayoutMonth)
+            .Chunk(HorizontalMonths)
+            .Select(EnumerableExtensions.Transpose)
+            .Select(JoinLine)
+            .SelectMany(Identity)
+            .JoinToString(Environment.NewLine);
+
+    private static DateOnly NextDay(DateOnly day)
+        => day.AddDays(1);
+
+    private static DateOnly JanuaryFirst(int year)
+        => new(year, 1, 1);
```

Comment:
add argument labels

Reply:
Fixed in 9e90ff193883aab4acb6fbb4758a68b84cac8a23